yuvals2146/uniapp
# Add Tests and some crucial changes

What in this large pr:
1. change the .env files to .envrc to support direnv
2. add direnv (https://direnv.net/) 
3. add running scripts in bin folder:
   a. test
   b. run
   c. logger (helper)
4. add db commends in package json to help with prisma cmds
5. add cascade delete to positionInfo table
6. cleaning the few savePositionDataSQL remains 
7. add some try/catch and fix alerts to getPositionData
8. add condition to NOT log if in test for readable test results 
9. create mock file for test objects
10. create factory file for test help functions
11. add yarn.lock

Few things: 
1. form now on we should use yarn instead of npm 

> The major difference between NPM and Yarn comes in terms of security performance. While NPM installs packages sequentially, Yarn performs parallel installation resulting in better speed and performance. NPM has tried to fix vulnerabilities, but still, Yarn is considered more secure than NPM.

2. to run our code lets start use `. bin/start` and `.bin/test/` the bash script can hold more logic and conditions
3. form now on lets try to make our process test driven development and start from the test other wise we might neglect this part in our 'move ultra fast and break everything'

Let me know if anything is unclear (please do not hesitate to ask me for rename,change,rewrite or anything. everything here is open for changes ❤️ 🚀 🌔 )
